onnxruntime/core/flatbuffers/schema/ort_training_checkpoint.fbs

```diff
@@ -12,6 +12,9 @@ table ModuleState {
   // are empty. i.e. The tensors are treated as named entities
   // without any meaningful data.
   is_nominal_state:bool;
+  // Tensors use external data file
+  // TODO: use bool or external data filename?
+  has_external_data:bool;
```

Comment:
I think the bool is fine for now. There is no need to support multiple files for on-device training since ideally we would limit the number of artifacts needed to be moved onto the device. save_checkpoint also provides the option of specifying the checkpoint name for the user. In the future, if there's demand for it, we can add another field for external data file names.